numpy/_core/memmap.py

```diff
@@ -250,7 +250,7 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
                 size = bytes // _dbytes
                 shape = (size,)
             else:
-                if type(shape) not in (tuple, list):
+                if type(shape) not in {tuple, list}:
```

Comment:
perhaps it's out of scope, but since I don't see why subclasses shouldn't be considered here, so we might as well write it as

```suggestion
                if not isinstance(shape, tuple | list):
```